libcalico-go/lib/ipam/ipam.go

```diff
@@ -1806,24 +1807,57 @@ func (c ipamClient) GetAssignmentAttributes(ctx context.Context, addr net.IP) (m
 // has been set, returns a default configuration with StrictAffinity disabled
 // and AutoAllocateBlocks enabled.
 func (c ipamClient) GetIPAMConfig(ctx context.Context) (config *IPAMConfig, err error) {
-	obj, err := c.client.Get(ctx, model.IPAMConfigKey{}, "")
-	if err != nil {
-		if _, ok := err.(cerrors.ErrorResourceDoesNotExist); ok {
-			// IPAMConfig has not been explicitly set.  Return
-			// a default IPAM configuration.
-			config = &IPAMConfig{
-				AutoAllocateBlocks: true,
-				StrictAffinity:     false,
-				MaxBlocksPerHost:   0,
+	var obj *model.KVPair
+
+	var retries int
+	maxRetry := 5
+
+	// First, trying to get IPAM config (create if not exists) for 5 times.
+	for retries = 1; retries < maxRetry; retries++ {
+		if ctx.Err() != nil {
+			return nil, ctx.Err()
+		}
+
+		obj, err = c.client.Get(ctx, model.IPAMConfigKey{}, "")
+		if err != nil {
+			// Create the default config if it doesn't already exist.
+			if _, ok := err.(cerrors.ErrorResourceDoesNotExist); ok {
+				// Write to datastore.
+				new := &model.KVPair{
+					Key: model.IPAMConfigKey{},
+					Value: &model.IPAMConfig{
+						StrictAffinity:     false,
+						AutoAllocateBlocks: true,
+						MaxBlocksPerHost:   0,
+					},
+				}
+
+				obj, err = c.client.Create(ctx, new)
+				if err != nil {
+					if _, ok := err.(cerrors.ErrorResourceAlreadyExists); ok {
+						log.Info("Failed to create global IPAM config; another node got there first.")
+						time.Sleep(1 * time.Second)
+						continue
+					}
+					log.WithError(err).Errorf("Error creating IPAM config")
```

Comment:
We should make sure our CNI plugin and calico/node RBAC is updated to include `create` permissions on this, if it doesn't have it already.